webapp/crashstats/signature/tests/test_views.py

```diff
@@ -2,506 +2,524 @@
-                        {
-                            "date": "2017-01-31T23:12:57",
-                            "uuid": "e325b443-1b51-402b-b2c5-ccd630220208",
-                            "product": "WaterWolf",
-                            "version": "1.0",
-                            "platform": "Linux",
-                            "build_id": None,
-                        },
-                        {
-                            "date": "2017-01-31T23:12:57",
-                            "uuid": "2e982308-dc57-41a1-b997-b56f40220208",
-                            "product": "WaterWolf",
-                            "version": "1.0",
-                            "platform": "Linux",
-                            "build_id": None,
-                        },
-                    ],
-                    "total": 4,
-                }
-                results["hits"] = self.only_certain_columns(
-                    results["hits"], params["_columns"]
-                )
-                return results
-
-            return {"hits": [], "total": 0}
-
-        SuperSearchUnredacted.implementation().get.side_effect = mocked_supersearch_get
+    def test_signature_reports(self, client, db, es_helper):
+        def build_crash_data(crash_id, **params):
+            data = {
+                "date_processed": date_to_string(date_from_ooid(crash_id)),
+                "uuid": crash_id,
+                "product": "Firefox",
+                "signature": TEST_SIGNATURE,
+                "version": "1.0",
+                "os_name": "Linux",
+            }
+            data.update(params)
+            return data
+
+        crash1_id = create_new_ooid()
+        crash1 = build_crash_data(
+            crash1_id,
+            build=888981,
+            cpu_info="FakeAMD family 20 model 42",
+        )
+        crash2_id = create_new_ooid()
+        crash2 = build_crash_data(
+            crash2_id,
+            build=888981,
+            cpu_info="AuthenticAMD family 20 model 1",
+        )
+        crash3_id = create_new_ooid()
+        crash3 = build_crash_data(crash3_id, build=None)
+        crash4_id = create_new_ooid()
+        crash4 = build_crash_data(crash4_id, build=None)
+
+        for crash_data in [crash1, crash2, crash3, crash4]:
+            es_helper.index_crash(
+                raw_crash={}, processed_crash=crash_data, refresh=False
+            )
+        es_helper.refresh()
+
+        start_date, end_date = get_date_range(crash1_id)
+        crash1_date_processed = date_from_ooid(crash1_id).strftime("%Y-%m-%d %H:%M:%S")
 
         # Test with no results.
         url = reverse("signature:signature_reports")
-        response = self.client.get(
-            url, {"signature": DUMB_SIGNATURE, "date": "2012-01-01"}
+        response = client.get(
+            url,
+            {
+                "signature": "wrong signature",
+                "product": "Firefox",
+                "date": [f">={start_date}", f"<{end_date}"],
```

Comment:
Now that we're not mocking everything out, we discover that passing dates is important, so this adds dates to all the supersearch filters.